RecipeMatch/API/APIManager.m

```diff
@@ -17,32 +23,230 @@ + (instancetype)shared {
     dispatch_once(&onceToken, ^{
         sharedManager = [[self alloc] init];
     });
+    
     return sharedManager;
 }
 
-- (void)getRecipes:(void(^)(NSArray *recipes, NSError *error))completion{
+- (instancetype)init {
+    NSString *path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
+    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
+
+    self.app_id = [dict objectForKey: @"app_id"];
+    self.app_key = [dict objectForKey: @"app_key"];
+    
+    return self;
+}
+
+// get initial array of recipes for home feed from recipe API
+- (void)getRecipesWithPreferences:( NSString * _Nullable )preferences andCompletion: (void (^)(NSMutableArray *recipe, NSError *error))completion{
     //Do any additional setup after loading the view.
-    NSURL *url = [NSURL URLWithString:@"https://api.edamam.com/api/recipes/v2?type=public&q=apple&app_id=00fb2355&app_key=1020f34ec9260531e5ad653a90e2d111"];
+    
+    NSString *apiString = @"https://api.edamam.com/api/recipes/v2?type=public&q=fruit&app_id=";
+    apiString = [apiString stringByAppendingString:self.app_id];
+    apiString = [apiString stringByAppendingString:@"&app_key="];
+    apiString = [apiString stringByAppendingString:self.app_key];
+    if(preferences){
+        apiString = [apiString stringByAppendingString: preferences];
+    }
+    NSURL *url = [NSURL URLWithString:apiString];
+    
     NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
     NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
     NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
            if (error != nil) {
                NSLog(@"%@", [error localizedDescription]);
                completion(nil, error);
-               
+
            }
            else {
                NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
 
                NSMutableArray *recipes = dataDictionary[@"hits"];
    
                completion(recipes, nil);
-
            };
 }];
     [task resume];
 }
 
+// get specific recipe by id from recipe API
+- (void)getRecipeWithId:( NSString * _Nullable )recipeId andCompletion: (void (^)(NSDictionary *recipe, NSError *error))completion{
+    //Do any additional setup after loading the view.
```

Comment:
@ellabuysse Let's make sure the formatting is correct, try to avoid extra spaces & comments. I am not sure if there is a linter in github that you can use, to point out possible formatting issues. I will share with you code formatting guidelines in our 1:1.